Merge datab to main

sujanyasrinath opened the pull request:
Added Database schemas for User and Items for use in connecting the programs backend with our MongoDB database

camilleparadis committed:
login pages update interface

camilleparadis committed:
Merge branch 'login-pages'

jkailee committed:
Merge pull request #31 from what2keep/main

updating your branch

jkailee committed:
Merge pull request #33 from what2keep/main

updating login pages with main

jkailee committed:
register now makes post call with User JSON

rhtakaha committed:
added post for adding user (just for testing)

jkailee committed:
Merge pull request #34 from what2keep/main

added post for adding user (just for testing)

rhtakaha committed:
got it to work

jkailee committed:
debugging port for azure

jkailee committed:
added .deployment

jkailee committed:
updated .deployment

rhtakaha committed:
added the email, password, and name to the result

jkailee committed:
updated project path in .deployment

jkailee committed:
Update README

rhtakaha committed:
added create and read for users

rhtakaha committed:
added the UD and fixed some linting errors

rhtakaha committed:
added full CRUD for Users
testing is almost complete
Not sure how to test for a read of all users to fail except with no users which will require the memory testing setup which I can do Friday/ over break

rhtakaha committed:
added a mongoose.disconnect() so that jest would stop auto running forever

jkailee committed:
Update README

jkailee committed:
Rename README to README.md

rhtakaha commented:
I updated the "/users" post call so make sure not to override that.
Other than that it seems fine.

camilleparadis reviewed (approved):
LGTM! Definitely double-check not to overwrite, as Ryan mentioned.

rhtakaha reviewed (approved):
I updated the "/users" post call so make sure not to override that.
Other than that it seems fine.

jkailee commented:
LGTM, all checks are passing

jkailee reviewed (approved):
LGTM

camilleparadis committed:
fixed font problems

camilleparadis committed:
added azure link instead of local ip

rhtakaha committed:
Merge branch 'main' into CRUD

rhtakaha committed:
Merge pull request #43 from what2keep/CRUD

User CRUD

jkailee committed:
Update README.md

jkailee committed:
renamed workflow

jkailee committed:
changed workflow name

jkailee committed:
checking if database is failing

jkailee committed:
Update dbCalls.js

jkailee committed:
added the URI to workflow

jkailee committed:
Create LICENSE

jkailee committed:
Update README.md

rhtakaha committed:
added the working memory db for testing
98.27% coverage because the unit tests connect to the memory instead of using the code to connect to the cloud
not sure what to do about that. Can ask BJ when we get back Monday

camilleparadis committed:
Merge pull request #41 from what2keep/login-pages

Login pages

camilleparadis committed:
Merge branch 'main' of https://github.com/what2keep/what2keep

rhtakaha committed:
Merge branch 'main' into localTests

rhtakaha committed:
Merge pull request #46 from what2keep/localTests

added the working memory db for testing

camilleparadis committed:
interface for items, stuff, etc

rhtakaha committed:
setup Items.js to work with local testing and for use

rhtakaha committed:
should be the core CRUD for items, need to write the tests and such

rhtakaha committed:
fully covered tests for item CRUD
pretty sure we also need a "usage" field which would be used for sorting and determineing "what2keep" but can add that fairly simply

rhtakaha committed:
added name and usage and ensured api calls work

rhtakaha committed:
Merge pull request #48 from what2keep/itemsCRUD

Items crud

rhtakaha committed:
a few changes to api_calls to facilitate login/registeration

jkailee committed:
Merge pull request #45 from what2keep/readme-editing

Readme

rhtakaha committed:
added ability to get the items for a user by
1) category
2) items with less than or equal to a given usage number

rhtakaha committed:
a few more comments/ clean up

rhtakaha committed:
Merge pull request #49 from what2keep/specialRead

Special read

rhtakaha committed:
cleaned up the backend

camilleparadis committed:
Merge pull request #50 from what2keep/main

main into branch

jkailee committed:
Update README.md

jkailee committed:
Update README.md

camilleparadis committed:
git st

camilleparadis committed:
not complete, added a few things to pass info between screens and started up front end backend link

jkailee committed:
update deployment

deployment only runs on pushes, and at midnight every day

jkailee committed:
Update main_what2keep.yml

will only run deployment when branch is merged

camilleparadis committed:
a bit of interface for stats

rhtakaha committed:
I believe that addItems should work
will need login pages/register
to send the userId to stuff and then to AddItem to see

rhtakaha committed:
got login page to work
BUT I need to modify the get users api call to accept password too so if checks email and password

camilleparadis committed:
Merge branch 'home-page-and-items' of https://github.com/what2keep/what2keep into home-page-and-items

rhtakaha committed:
completed login
added to apicalls email+password login (fully tested)
moved the userId through the screens to AddItems which now works

rhtakaha committed:
added some more routing
EditItem almost done but not sure since ViewItem can't send itemId yet

camilleparadis committed:
lil changes to stuff and home

camilleparadis committed:
Merge branch 'home-page-and-items' of https://github.com/what2keep/what2keep into home-page-and-items

Commit:
Stats page

jkailee committed:
Revert "Stats page"

This reverts commit 80c0430ca6853dec3c9db3829d579ddba0bb30a1.

jkailee committed:
Update main_what2keep.yml

camilleparadis committed:
stuff page and view item w/o testing

camilleparadis committed:
new

camilleparadis committed:
in root

camilleparadis committed:
m

rhtakaha committed:
testing

rhtakaha committed:
Merge branch 'home-page-and-items' of https://github.com/camilleparadis/what2keep into home-page-and-items

rhtakaha committed:
fixing up stuff and view

rhtakaha committed:
working images for view and stuff

rhtakaha committed:
added images to the item thumbnails

rhtakaha committed:
removed some testing console logs

rhtakaha committed:
fixed EditItem to work
made a separate file to hold the ip/url so easier to change between all at once

rhtakaha committed:
commented out a console log

camilleparadis committed:
added android supported font n safe area view. cant run on ios so not sure if its working on ryans

camilleparadis committed:
added android supported fonts n safe area view

camilleparadis committed:
delete item, will need to debug ryans, cant run on mine

rhtakaha committed:
changed ip

rhtakaha committed:
Merge branch 'home-page-and-items' of https://github.com/camilleparadis/what2keep into home-page-and-items

rhtakaha committed:
delete items works

rhtakaha committed:
fixed get items for no items to not be an error

rhtakaha committed:
made the stuff page refresh after edit/add/delete automatically
cleaned up code a bit

rhtakaha committed:
got rid of a console log

rhtakaha committed:
Merge pull request #52 from what2keep/home-page-and-items

Home page and items

jkailee committed:
Update README.md

camilleparadis committed:
changed fonts back to ios supported

jkailee committed:
interfaces with azure deployment instead of local

jkailee committed:
Merge pull request #53 from what2keep/home-page-and-items

Home page and items

jkailee committed:
Update README.md
